### PR TITLE
Redirect fixes

### PR DESCRIPTION
**Pull request by gianm**
* Fix overlord redirect's ability to detect that there is no leader.
  * Change the 302 to 307 so user agents are more likely to preserve the method (POST -> POST instead of POST -> GET)

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

File: indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordRedirectInfo.java [hunk 1]
```diff
@@ -45,8 +47,17 @@ public boolean doLocal()
   @Override
   public URL getRedirectURL(String queryString, String requestURI)
   {
+    if (!requestURI.startsWith("/")) {
```
Comment (xvrl) [inline indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordRedirectInfo.java]: is this even necessary given that we use new URI(...) ?

Comment (gianm) [inline indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordRedirectInfo.java]: no

Comment (gianm) [inline indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordRedirectInfo.java]: removed

**Pull request merged**